test/Microsoft.AspNetCore.Mvc.Core.Test/Authorization/AuthorizeFilterTest.cs

```diff
@@ -333,7 +348,59 @@ public void InvalidUser()
             Assert.NotNull(authorizationContext.Result);
         }
 
-        private Filters.AuthorizationFilterContext GetAuthorizationContext(
+        [Fact]
+        public void CreateInstance_ReturnsSelfIfPolicyIsSet()
+        {
+            // Arrange
+            var authorizeFilter = new AuthorizeFilter(new AuthorizationPolicyBuilder()
+                .RequireAssertion(_ => true)
+                .Build());
+            var factory = (IFilterFactory)authorizeFilter;
+
+            // Act
+            var result = factory.CreateInstance(new ServiceCollection().BuildServiceProvider());
+
+            // Assert
+            Assert.Same(authorizeFilter, result);
+        }
+
+        [Fact]
+        public void CreateInstance_ReturnsSelfIfPolicyProviderIsSet()
+        {
+            // Arrange
+            var authorizeFilter = new AuthorizeFilter(new AuthorizationPolicyBuilder()
+                .RequireAssertion(_ => true)
+                .Build());
+            var factory = (IFilterFactory)authorizeFilter;
+
+            // Act
+            var result = factory.CreateInstance(new ServiceCollection().BuildServiceProvider());
+
+            // Assert
+            Assert.Same(authorizeFilter, result);
+        }
+
+        [Fact]
+        public void CreateInstance_ReturnsNewPolicyIfPolicyAndPolicyProviderIsNotSet()
+        {
+            // Arrange
+            var authorizeFilter = new AuthorizeFilter(new[] { new AuthorizeAttribute() });
+            var factory = (IFilterFactory)authorizeFilter;
+            var serviceProvider = new ServiceCollection()
+                .AddOptions()
+                .AddAuthorization()
+                .BuildServiceProvider();
+
+            // Act
+            var result = factory.CreateInstance(serviceProvider);
```

Comment:
I mean the filter instance on one more call to CreateInstance should create the same instance of the filter (I ask this because the other tests here short circuit in the if condition and do not go through DI)